src/pde_solve.jl

```diff
@@ -59,5 +66,87 @@ function DiffEqBase.solve(
 
     Flux.train!(loss, ps, data, opt; cb = cb)
 
-    save_everystep ? iters : u0(X0)[1]
+
+    if give_limit == false
+        save_everystep ? iters : u0(X0)[1]
+    else
+        A = prob.kwargs.data.A
+        u_domain = prob.kwargs.data.u_domain
+
+        ## UPPER LIMIT
+        m1 = trajectories_upper
+        sdeProb = SDEProblem(μ , σ , X0 , prob.tspan)
+        ensembleprob = EnsembleProblem(sdeProb)
+        sim = solve(ensembleprob, sdealg, ensemblealg, dt=dt,trajectories=m1,adaptive=false)
+        function sol_high()
+            Uo = []
+            for u in sim.u
+                xsde = u.u
+                U = g(xsde[end])
+                u = u0(X0)[1]
+                for i in length(ts):-1:3
+                    t = ts[i]
+                    _σᵀ∇u = σᵀ∇u[i-1](xsde[i-1])
+                    dW = sqrt(dt)*randn(d)
+                    U = U .+ f(xsde[i-1], U, _σᵀ∇u, p, t)*dt .- _σᵀ∇u'*dW
+                end
+                Uo = vcat(Uo , U)
+            end
+            Uo
+        end
+
+        loss_() = sum(sol_high())/m1
+
+        ps = Flux.params(u0, σᵀ∇u...)
+        cb = function ()
+            l = loss_()
+            true && println("Current loss is: $l")
+            l < 1e-6 && Flux.stop()
+        end
+        dataS = Iterators.repeated((), maxiters_upper)
+        Flux.train!(loss_, ps, dataS, ADAM(0.01); cb = cb)
+        u_high = loss_()
+        println(u_high)
+        ##Lower Limit
+
+        # Function to precalculate the f values over the domain
+        function give_f_matrix(X,urange,σᵀ∇u,p,t)
+            a = []
+            for u in urange
+                a = vcat(a , f(X,u,σᵀ∇u,p,t))
+            end
+            return a
+        end
+        #The legendre transform that uses the precalculated f values.
+        function legendre_transform(f_matrix , a , urange)
+            le = a.*(collect(urange)) .- f_matrix
+            return maximum(le)
+        end
+
+        m2 = trajectories_lower
+        function sol_low()
+            map(1:m2) do j
+                u = u0(X0)[1]
+                X = X0
+                I = 0.0
```

Comment:
```suggestion
                I = zero(eltype(u))
```

and the same on all other zeros. Many times they should be Float32 and not Float64